Source/GTMSessionFetcher.m

```diff
@@ -551,8 +551,11 @@ - (void)beginFetchMayDelay:(BOOL)mayDelay
                            fetchRequest);
     }
 #endif
+    // If priorSessionIdentifier is allowed to stay non-nil, a background session can
```

Comment:
Would the previous try not have been background if it was a data url?  And you can't change the url after the start, so what would have gotten it into this state?

Reply:
The concern is that priorSessionIdentifier is what's [checked below](https://github.com/google/gtm-session-fetcher/blob/96b3d35b30a983f152d4cc11db0adb45adf363eb/Source/GTMSessionFetcher.m#L634), so if a data URL is provided but a session identifier were to be set, we could still get in a situation where we attempt to create a background session if `useBackgroundSession` is true. Also, by changing `useBackgroundSession` and not `usingBackgroundSession` we're in the same boat.